dali/test/python/test_operator_random_bbox_crop.py

```diff
@@ -0,0 +1,230 @@
+        return False
+    return True
+
+def filter_by_centroid(crop_anchor, crop_shape, bboxes):
+    ndim = len(crop_shape)
+    nboxes = bboxes.shape[0]
+    indexes = []
+    for i in range(nboxes):
+        bbox = bboxes[i]
+        centroid = [0.5 * (bbox[d] + bbox[ndim + d]) for d in range(ndim)]
+        if crop_contains(crop_anchor, crop_shape, centroid):
+            indexes.append(i)
+        filtered_boxes = np.array(bboxes[indexes, :])
+    return filtered_boxes
+
+def map_box(bbox, crop_anchor, crop_shape):
+    ndim = int(len(bbox) / 2)
+    assert len(crop_anchor) == ndim
+    assert len(crop_shape) == ndim
+    new_bbox = np.array(bbox)
+    for d in range(ndim):
+        c_start = crop_anchor[d]
+        c_end = crop_anchor[d] + crop_shape[d]
+        b_start = bbox[d]
+        b_end = bbox[ndim+d]
+        rel_extent = c_end - c_start
+        n_start = (max(c_start, b_start) - c_start) / rel_extent
+        n_end = (min(c_end, b_end) - c_start) / rel_extent
+        new_bbox[d] = max(0.0, min(1.0, n_start))
+        new_bbox[ndim + d] = max(0.0, min(1.0, n_end))
+    return new_bbox
+
+def check_processed_bboxes(crop_anchor, crop_shape, original_boxes, processed_boxes):
+    filtered_boxes = filter_by_centroid(crop_anchor, crop_shape, original_boxes)
+    assert(len(original_boxes) >= len(filtered_boxes))
+    assert(len(filtered_boxes) == len(processed_boxes))
+    nboxes = len(filtered_boxes)
+    for i in range(nboxes):
+        box = filtered_boxes[i]
+        processed_box = processed_boxes[i]
+        expected_box = map_box(box, crop_anchor, crop_shape)
+        assert(np.allclose(expected_box, processed_box, atol=1e-6))
+
+def check_crop_dims_variable_size(anchor, shape, scaling, aspect_ratio):
+    ndim = len(shape)
+    for d in range(ndim):
+        assert(anchor[d] >= 0.0 and anchor[d] <= 1.0)
+        assert(anchor[d] + shape[d] > 0.0 and anchor[d] + shape[d] <= 1.0)
+        assert(shape[d] >= scaling[0] and shape[d] <= scaling[1])
+        ar = shape[0] / shape[1]
+        assert(ar >= aspect_ratio[0] and ar <= aspect_ratio[1])
+
+def check_crop_dims_fixed_size(anchor, shape, expected_crop_shape, input_shape):
+    ndim = len(shape)
+    for d in range(ndim):
+        assert(anchor[d] >= 0.0 and anchor[d] <= input_shape[d])
+        assert shape[d] == expected_crop_shape[d], "{} != {}".format(shape, expected_crop_shape)
+        assert(anchor[d] + shape[d] > 0.0 and anchor[d] + shape[d] <= input_shape[d])
+
+bbox_2d_ltrb_1 = [0.0, 0.0, 0.9, 0.9]
+bbox_2d_ltrb_2 = [0.1, 0.1, 0.99, 0.99]
+bbox_2d_ltrb_3 = [0.3, 0.3, 0.5, 0.5]
+bbox_3d_ltrb_1 = [0.5, 0.5, 0.5, 0.7, 0.7, 0.7]
+bbox_3d_ltrb_2 = [0.1, 0.1, 0.1, 0.6, 0.6, 0.6]
+bbox_3d_ltrb_3 = [0.4, 0.4, 0.4, 0.9, 0.9, 0.9]
+
+def get_bounding_boxes(ndim, nboxes, produce_labels=True):
+    assert(nboxes == 3)
+    if ndim == 2:
+        boxes = [np.array([bbox_2d_ltrb_1, bbox_2d_ltrb_2, bbox_2d_ltrb_3], dtype=np.float32)]
```

Comment:
please add another a second item to the batch and preferably another (different) batch.
assuming your boxes are A,B,C, you could have:
```
batch1 = [ [A, B, C], [B, A] ]
batch2 = [ [C], [A, C, B] ]
```
This would check that:
1. Multiple samples are actually handled
2. Sample indexing is right
3. Different size of box list within batch is handled
4. The operator setup works each time, not just the first
5. The internal state doesn't prevent the operator from handling a completely different batch.